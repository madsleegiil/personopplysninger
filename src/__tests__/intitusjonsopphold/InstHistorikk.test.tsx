import "@testing-library/jest-dom";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";
import InstHistorikkView from "../../pages/institusjonsopphold/historikk/InstHistorikkView";
import instInfo from "../../clients/apiMock/app/fetch/inst-info.json";
import { render } from "@testing-library/react";
import moment from "moment-timezone";

moment.tz.setDefault("GMT+1");

jest.mock("react-router-dom", () => ({
  Link: () => <div />,
  useLocation: () => ({
    pathname: "pathname",
  }),
}));

jest.mock("nav-frontend-hjelpetekst");

describe("InstHistorikkView", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <IntlProvider locale={"nb"} messages={nbMessages}>
        <InstHistorikkView instInfo={instInfo} />
      </IntlProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
