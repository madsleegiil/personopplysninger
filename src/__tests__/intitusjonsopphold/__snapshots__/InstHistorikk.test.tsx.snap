// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstHistorikkView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arbeidsforhold__disclaimer"
  >
    <div
      class="alertstripe alertstripe--info"
    >
      <span
        class="alertstripe__ikon"
      >
        <span
          class="sr-only"
        >
          info
        </span>
        <svg
          aria-label="info-ikon"
          focusable="false"
          height="1.5em"
          kind="info-sirkel-fyll"
          role="img"
          viewBox="0 0 24 24"
          width="1.5em"
        >
          <g
            fill="none"
          >
            <path
              d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
              fill="#337C9B"
            />
            <path
              d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
              fill="#FFF"
            />
          </g>
        </svg>
      </span>
      <div
        class="typo-normal alertstripe__tekst"
      >
        <p
          class="typo-normal"
        >
          Her kan du se opplysninger om opphold på alders- og sykehjem, helseinstitusjoner og fengsel fra 1.januar 2017. Oppdager du feil, må du kontakte institusjonen direkte.
        </p>
      </div>
    </div>
    <div
      class="inst__tabell"
    >
      <div
        class="historikk__flex-rad inst__head"
      >
        <div
          class="historikk__flex-kolonne"
        >
          <p
            class="typo-element"
          >
            Periode
          </p>
        </div>
        <div
          class="historikk__flex-kolonne"
        >
          <p
            class="typo-element"
          >
            Institusjon
          </p>
        </div>
      </div>
      <div>
        <div
          class="historikk__flex-rad"
          style="transition-delay: 50ms; font-weight: normal;"
        >
          <div
            class="historikk__flex-kolonne historikk__heading"
          >
            <time
              datetime="1514502000000"
            >
              29.12.2017
            </time>
             - 
            <time
              datetime="1545001200000"
            >
              17.12.2018
            </time>
          </div>
          <div
            class="historikk__flex-kolonne"
          >
            <div />
          </div>
        </div>
        <div
          class="historikk__flex-rad"
          style="transition-delay: 65ms; font-weight: normal;"
        >
          <div
            class="historikk__flex-kolonne historikk__heading"
          >
            <time
              datetime="1514502000000"
            >
              29.12.2017
            </time>
             - 
            <time
              datetime="1608418800000"
            >
              20.12.2020
            </time>
          </div>
          <div
            class="historikk__flex-kolonne"
          >
            <div />
          </div>
        </div>
        <div
          class="historikk__flex-rad"
          style="transition-delay: 80ms; font-weight: normal;"
        >
          <div
            class="historikk__flex-kolonne historikk__heading"
          >
            <time
              datetime="1497304800000"
            >
              13.06.2017
            </time>
             - 
            <time
              datetime="1505426400000"
            >
              15.09.2017
            </time>
          </div>
          <div
            class="historikk__flex-kolonne"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
    <div
      class="inst__kilde"
    >
      <div
        class="kilde__container"
      >
        <div
          class="kilde__tekst"
        >
          <p
            class="typo-undertekst"
          >
            Kilde: Den enkelte institusjon
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
